website/pages/docs/going-to-production.mdx

```diff
@@ -124,3 +134,296 @@ export default [
   }
 }
 ```
+
+## Secure your schema
+
+GraphQL gives clients a lot of flexibility, which can be a strength or a liability depending on
+how it's used. In production, it's important to control how much of your schema is exposed
+and how much work a single query is allowed to do.
+
+Common strategies for securing a schema include:
+
+- Disabling introspection for some users
+- Limiting query depth or cost
+- Enforcing authentication and authorization
+- Applying rate limits
+
+These techniques can help protect your server from accidental misuse or intentional abuse.
+
+### Control schema introspection
+
+Introspection lets clients query the structure of your schema, including types and fields. While
+helpful during development, it may expose internal details you don't want to reveal in
+production.
+
+You can disable introspection in production, or only for unauthenticated users:
+
+```js
+import { validate, specifiedRules, NoSchemaIntrospectionCustomRule } from 'graphql';
+
+const validationRules = isPublicRequest
+  ? [...specifiedRules, NoSchemaIntrospectionCustomRule]
+  : specifiedRules;
+```
+
+Note that many developer tools rely on introspection to function properly. Use introspection
+control as needed for your tools and implementation.
+
+### Limit query complexity
+
+GraphQL allows deeply nested queries, which can be expensive to resolve. You can prevent this
+with query depth limits or cost analysis.
+
+The following example shows how to limit query depth:
+
+```js
+import depthLimit from 'graphql-depth-limit';
+
+const validationRules = [
+  depthLimit(10),
+  ...specifiedRules,
+];
+```
+
+Instead of depth, you can assign each field a cost and reject queries that exceed a total budget.
+Tools like [`graphql-cost-analysis`](https://github.com/pa-bru/graphql-cost-analysis) can help.
+
+### Require authentication and authorization
+
+GraphQL doesn't include built-in authentication. Instead, you can attach user data to the request
+using middleware, then enforce authorization in your resolvers:
+
+```js
+function requireRole(role, resolver) {
+  return (parent, args, context, info) => {
+    if (context.user?.role !== role) {
+      throw new Error('Not authorized');
+    }
+    return resolver(parent, args, context, info);
+  };
+}
+```
+
+For more details, see the [Authentication and Middleware](./authentication-and-express-middleware/) guide.
+
+### Apply rate limiting
+
+To prevent abuse, you can limit how often clients send queries. Basic rate limiting can be added
```

Comment:
Should we link to https://github.com/teamplanes/graphql-rate-limit/blob/master/README.md#field-config instead